Blueprints/BlueprintDefinitions/vs2022/AspNetCoreWebAPI-Image-FSharp/template/src/BlueprintBaseName.1/Readme.md

```diff
@@ -0,0 +1,101 @@
+* web.config - used for local development.
+* Controllers\ValuesController.fs - example Web API controller
+
+You may also have a test project depending on the options selected.
+
+### Configuring for API Gateway HTTP API ###
+
+API Gateway supports the original REST API and the new HTTP API. In addition HTTP API supports 2 different
+payload formats. When using the 2.0 format the base type of `LambdaEntryPoint` must be `Amazon.Lambda.AspNetCoreServer.APIGatewayHttpApiV2ProxyFunction`.
+For the 1.0 payload format the base type is the same as REST API which is `Amazon.Lambda.AspNetCoreServer.APIGatewayProxyFunction`.
+**Note:** when using the `AWS::Serverless::Function` CloudFormation resource with an event type of `HttpApi` the default payload
+format is 2.0 so the base type of `LambdaEntryPoint` must be `Amazon.Lambda.AspNetCoreServer.APIGatewayHttpApiV2ProxyFunction`.
+
+### Configuring for Application Load Balancer ###
+
+To configure this project to handle requests from an Application Load Balancer instead of API Gateway change
+the base type of `LambdaEntryPoint` from `Amazon.Lambda.AspNetCoreServer.APIGatewayProxyFunction` to 
+`Amazon.Lambda.AspNetCoreServer.ApplicationLoadBalancerFunction`.
+
+## Packaging as a Docker image.
+
+This project is configured to package the Lambda function as a Docker image. The default configuration for the project and the Dockerfile is to build 
+the .NET project on the host machine and then execute the `docker build` command which copies the .NET build artifacts from the host machine into 
+the Docker image. 
+
+The `--docker-host-build-output-dir` switch, which is set in the `aws-lambda-tools-defaults.json`, triggers the 
+AWS .NET Lambda tooling to build the .NET project into the directory indicated by `--docker-host-build-output-dir`. The Dockerfile 
+has a **COPY** command which copies the value from the directory pointed to by `--docker-host-build-output-dir` to the `/var/task` directory inside of the 
+image.
+
+Alternatively the Docker file could be written to use [multi-stage](https://docs.docker.com/develop/develop-images/multistage-build/) builds and 
+have the .NET project built inside the container. Below is an example of building .NET 5 project inside the image.
+
+```dockerfile
+FROM public.ecr.aws/lambda/dotnet:5.0 AS base
+
+FROM mcr.microsoft.com/dotnet/sdk:5.0-buster-slim as build
+WORKDIR /src
+COPY ["BlueprintBaseName.1.csproj", "BlueprintBaseName.1/"]
+RUN dotnet restore "BlueprintBaseName.1/BlueprintBaseName.1.csproj"
+
+WORKDIR "/src/BlueprintBaseName.1"
+COPY . .
+RUN dotnet build "BlueprintBaseName.1.csproj" --configuration Release --output /app/build
+
+FROM build AS publish
+RUN dotnet publish "BlueprintBaseName.1.csproj" \
+            --configuration Release \ 
+            --runtime linux-x64 \
+            --self-contained false \ 
+            --output /app/publish \
+            -p:PublishReadyToRun=true  
```

Comment:
why not using <PublishReadyToRun>true</PublishReadyToRun> in csproj or the other way around. 

Currently we have two ways of doing this.

Reply:
This is not a new change but pre .NET 6 we would avoid putting it directly in the csproj file because that would make it be attempted to use for all publish commands even when you are not on Linux.